clients/geth/specular/rollup/services/base.go

```diff
@@ -65,6 +61,26 @@ func (b *BaseService) Chain() *core.BlockChain {
 	return b.Eth.BlockChain()
 }
 
+
+func (b *BaseService) GetL1RollupBlock(ctx context.Context, start uint64) uint64 {
```

Comment:
this should be something like
```suggestion
func (b *BaseService) GetLatestL1BlockForL2Block(ctx context.Context, l2Block uint64, start uint64) uint64 {
```